### ISSUE TITLE
[Fun task]: Add React tracing

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/jaeger-ui#1307]
### Requirement

Understand and visualize the behavior of Jaeger UI via traces

### Problem

Currently we only start traces at the query service.

### Proposal

Perhaps follow suggestions in the blog?

https://developers.redhat.com/articles/2023/03/22/how-enable-opentelemetry-traces-react-applications

### Open questions

How can we send trace data from the browser to Jaeger backend? Should the query-service implement a dedicated endpoint where frontend can report traces?

**Comment by psk001** [comment]
Hi @yurishkuro ,  is it up for grab can I start on it ?

**Comment by yurishkuro** [comment]
@psk001 yes

**Comment by psk001** [comment]
Thanks @yurishkuro. I followed the mentioned article and was able to get traces using [jaeger all-in-one](https://www.jaegertracing.io/docs/1.47/getting-started/#all-in-one) For the exporter, a URL is required to send traces to collector
```javascript
const collectorExporter = new OTLPTraceExporter({
// url:""
  headers: {}
});
```
Without a given value, it defaults to localhost. Is there a collector URL that can be added here?

**Comment by yurishkuro** [comment]
All-in-one listens on port 4318 for OTLP over HHTP.

**Comment by MAX-786** [comment]
is it up for grabs? :)

**Comment by yurishkuro** [comment]
Yes, but also see a previous unsuccessful PR #1627

**Comment by MAX-786** [comment]
yep, read it through and personally i think author got almost through with the implementation with few formatting/cleanup issues i suppose and one issue with adding a way to  disable the tracing.
soon I'll open up a draft PR if i got it right?

**Comment by yurishkuro** [comment]
@MAX-786 I think the main unresolved issue for me was how the traces from the UI should be exported. UI communicates with jaeger-query via an internal API. jaeger-query itself generates traces internally and exports them somewhere (in case of all-in-one it exports it back to itself over local port). So my assumption was that the zero-configuration option would be for the UI to export traces into an internal endpoint in the jaeger-query where it can handle them the same way as the the traces it itself generates.

A non-zero configuration option would be to make it the responsibility of the operator to provide an export URL for traces from the UI. It's certainly doable but makes the configuration much more complicated imo, as one has to potentially deal with additional CORS settings to allow the UI to talk to something else, etc.

**Comment by yurishkuro** [comment]
To be fair, we could start with the non-zero configuration and at least merge the React PR to capture the traces, to divide the problem. Then we can see if we can add a new endpoint to jaeger-query to accept traces from the UI (we already have an implementation of OTLP/HTTP endpoint, it's easy to reuse).

**Comment by vibhor-5** [comment]
Hey @yurishkuro should i have a go at this? I checked the last two closed PRs

**Comment by yurishkuro** [comment]
you can but I recommend writing an ADR first for how it would work, given the previous attempts and feedback.

**Comment by vibhor-5** [comment]
Yeaah fair enough, I'll get to back to you with an adr doc

**Comment by vibhor-5** [comment]
https://docs.google.com/document/d/1CzbCNc3kd6EU_OF6sc_ZAEzkoAU9xoOiG4ZbA22eJaQ/edit?usp=sharing

@yurishkuro please have a look

**Comment by yurishkuro** [comment]
The new backend endpoint is implemented, the UI changes are unblocked.

**Comment by pinankshah8-Aa** [comment]
Hi @yurishkuro, I’ve been reading ADR 0011 and the Phase 1 browser-tracing work in #4051. Rather than duplicating the already implemented Phase 1 work, I’d like to work on one of the deferred follow-ups: adding an `assistant.turn` root span around Jaeger Assistant runs.

I traced the current flow through the AG-UI `HttpAgent` and the existing OpenTelemetry fetch instrumentation. My proposed direction is to add tracing at the AG-UI middleware/run boundary so that each assistant run gets one `assistant.turn` span and the existing HTTP/fetch span becomes its child.

I also want the implementation not to depend on AG-UI’s current eager HTTP subscription behavior, so I would keep the downstream run subscription inside the turn span’s OpenTelemetry context and end the span when the run completes, errors, or is cancelled.

I would avoid recording prompt/response contents or other sensitive/high-cardinality data.

I plan to cover the behavior with tests for:
- one turn span per assistant run,
- parentage of the downstream request,
- completion/error/cancellation lifecycle.

Does this look like an appropriate next piece of #1307?

My previous PR #4441 has now merged, so I’m available to take this on if the direction looks right.

**Comment by pinankshah8-Aa** [comment]
Hi @yurishkuro, following up on my Sep 5 proposal. I re-read ADR-0011 and your earlier guidance in this thread to write an ADR first for this area.

I’ve drafted a small follow-up ADR focused only on the deferred AI Assistant named-operation tracing: one `assistant.turn` span per subscribed AG-UI run, the downstream subscription executed inside that OTel context so the existing fetch span becomes its child, and the span ending exactly once on complete/error/unsubscribe. It intentionally adds no prompt/response content, no new exporter/config, and no duplicate HTTP span.

Before I implement anything, would you prefer that I submit this as a formal ADR PR first for review? If yes, I’ll keep the implementation separate until the design is accepted.
